Beta_Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py

```diff
@@ -0,0 +1,354 @@
+        if not token:
+            raise RuntimeError('Failed to retrieve token')
+        return token
+
+    async def refresh_token_loop(self) -> None:
+        while True:
+            await sleep(self.expiry_time)
+            token = await self._get_access_token()
+            self.token = token
+            self.client.set_auth_headers(token)
+
+
+@asynccontextmanager
+async def init_refresh_token(
+        base_url: str,
+        client_id: str,
+        client_secret: str,
+        verify_ssl: bool,
+        proxy: bool
+) -> AsyncIterator[RefreshToken]:
+    """Initializes RefreshToken instance and authenticates with CrowdStrike Falcon.
+
+    Args:
+        base_url (str): CrowdStrike Falcon Cloud base URL.
+        client_id (str): CrowdStrike Falcon application ID.
+        client_secret (str): CrowdStrike Falcon application secret.
+        verify_ssl (bool): Whether the request should verify the SSL certificate.
+        proxy (bool): Whether to run the integration using the system proxy.
+
+    Yields:
+        AsyncIterator[RefreshToken]: RefreshToken instance initialized with client details.
+    """
+    refresh_token = RefreshToken(base_url, client_id, client_secret, verify_ssl, proxy)
+    task = create_task(refresh_token.refresh_token_loop())
+    yield refresh_token
+    task.cancel()
+
+
+async def long_running_loop(
+        base_url: str,
+        client_id: str,
+        client_secret: str,
+        stream: EventStream,
+        offset: int,
+        event_type: str,
+        verify_ssl: bool,
+        proxy: bool
+) -> None:
+    """Connects to a CrowdStrike Falcon stream and fetches events from it in a loop.
+
+    Args:
+        base_url (str): CrowdStrike Falcon Cloud base URL.
+        client_id (str): CrowdStrike Falcon application ID.
+        client_secret (str): CrowdStrike Falcon application secret.
+        stream (EventStream): CrowdStrike Falcon stream to fetch events from.
+        offset (int): Stream offset to start the fetch from.
+        event_type (str): Stream event type to fetch.
+        verify_ssl (bool): Whether the request should verify the SSL certificate.
+        proxy (bool): Whether to run the integration using the system proxy.
+
+    Returns:
+        None: No data returned.
+    """
+    async with init_refresh_token(base_url, client_id, client_secret, verify_ssl, proxy) as refresh_token:
+        stream.set_refresh_token(refresh_token)
+        async for event in stream.fetch_event(offset=offset, event_type=event_type):
+            event_metadata = event.get('metadata', {})
+            event_type = event_metadata.get('eventType', '')
+            event_offset = event_metadata.get('offset', '')
+            demisto.info(f'Fetching event with offset: {event_offset}')
+            incident_name = f'{event_type} - offset {event_offset}'
+            incident = [{
+                'name': incident_name,
+                'details': json.dumps(event)
```

Comment:
shouldn't this be in `rawJson`?

Reply:
no, because the `rawJSON` is being deleted in later stage by the server unless given a server config flag